libcontainer/cgroups/fs/memory.go

```diff
@@ -147,6 +141,40 @@ func memoryAssigned(cgroup *configs.Cgroup) bool {
 		cgroup.MemorySwappiness != -1
 }
 
+func getMemoryMappingData(path, name string) (map[string]uint64, error) {
+	data := make(map[string]uint64)
+
+	file, err := os.Open(filepath.Join(path, name))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	defer file.Close()
+
+	sc := bufio.NewScanner(file)
+	for sc.Scan() {
+		t, v, err := getCgroupParamKeyValue(sc.Text())
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse %s (%q) - %v", name, sc.Text(), err)
+		}
+		data[t] = v
+	}
+
+	return data, nil
+}
+
+func getMemoryStat(path string) (map[string]uint64, error) {
+	// Get stats from memory.stat.
+	return getMemoryMappingData(path, "memory.stat")
+}
+
+func getOomStat(path string) (map[string]uint64, error) {
```

Comment:
Personally `oom_control` feels more like `Status` rather than `Stats`.

On Sun, Dec 20, 2015 at 11:37 PM, Qiang Huang <notifications@github.com>
wrote:

> In libcontainer/cgroups/fs/memory.go
> <https://github.com/opencontainers/runc/pull/418#discussion_r48122236>:
>
> > +		t, v, err := getCgroupParamKeyValue(sc.Text())
> > +		if err != nil {
> > +			return nil, fmt.Errorf("failed to parse %s (%q) - %v", name, sc.Text(), err)
> > +		}
> > +		data[t] = v
> > +	}
> > +
> > +	return data, nil
> > +}
> > +
> > +func getMemoryStat(path string) (map[string]uint64, error) {
> > +	// Get stats from memory.stat.
> > +	return getMemoryMappingData(path, "memory.stat")
> > +}
> > +
> > +func getOomStat(path string) (map[string]uint64, error) {
>
> @vishh <https://github.com/vishh> Why do you think oomStat is not really
> a stat? From user's point of view, I think it surely is a stat which is
> also very concerned by users, and all this PR was inspired by help message
> of runc events
>
> $ runc events --help
> NAME:
>    events - display container events such as OOM notifications, cpu, memory, IO and network stats
>
> USAGE:
>    command events [command options] [arguments...]
>
> OPTIONS:
>    --interval "5s"      set the stats collection interval
>    --stats              display the container's stats then exit
>
> Which I noticed we don't have oom stats yet but it says we do. (I think
> original description OOM notifications stats was meat to be OOM stats)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencontainers/runc/pull/418/files#r48122236>.
>